File: exer-pratica-especializacao/livro/LivroVirtual.java
package livro;

import autor.Autor;

public class LivroVirtual extends Livro {

    public LivroVirtual(String id, String nome_livro, double valor_inicial, Autor autor_livro) {
        super(id, nome_livro, valor_inicial, autor_livro);
    }

    @Override
    public double calculaPrecoLivro() {

        return this.getValor_inicial() + this.getDiretos_autorais();
    
    }

    
}

File: exer-pratica-especializacao/livro/LivroVirtual.py
from autor.Autor import Autor
from livro.livro import Livro

class LivroVirtual(Livro):
    
    def __init__(self, id : str, nome_livro : str, 
    valor_inical : float, autor_livro : Autor):
        super().__init__(id, nome_livro, valor_inical, autor_livro)
    
    # override
    
    def calculaPrecoLivro(self):
        return self.valor_inicial + self.direitos_autorais